cmd/machine-os-builder/start.go

```diff
@@ -72,11 +75,12 @@ func getBuildController(ctx context.Context, cb *clients.Builder) (*build.Contro
 	buildClients := build.NewClientsFromControllerContext(ctrlCtx)
 	cfg := build.DefaultBuildControllerConfig()
 
-	if imageBuilderType == build.OpenshiftImageBuilder {
-		return build.NewWithImageBuilder(cfg, buildClients), nil
-	}
+	controllersToStart := []*build.Controller{}
 
-	return build.NewWithCustomPodBuilder(cfg, buildClients), nil
+	for range machineOSConfigs.Items {
+		controllersToStart = append(controllersToStart, build.NewWithCustomPodBuilder(cfg, buildClients))
```

Comment:
issue (blocking): If the intent is to have a single BuildController handle a single MachineOSConfig, how does a BuildController instance know which MachineOSConfig to handle? It doesn't look like the instantiated BuildController has any knowledge about a particular MachineOSConfig.